builder/builder.go

```diff
@@ -201,83 +201,58 @@ func (b *Builder) SignManifestMoM() error {
 	return nil
 }
 
-// UpdateRepo will fetch the clr-bundles for our configured Clear Linux version
-func (b *Builder) UpdateRepo(ver string, allbundles bool) {
-	// Make the folder to store all clr-bundles version
-	if _, err := os.Stat("clr-bundles"); err != nil {
-		if err = os.Mkdir("clr-bundles", 0777); err != nil {
-			fmt.Fprintf(os.Stderr, "ERROR: Failed to create clr-bundles: %s\n", err.Error())
-		}
-	}
+const (
+	upstreamBundlesBaseDir   = ".mixer/upstream-bundles/"
+	upstreamBundlesVerDir    = "clr-bundles-%s"
+	upstreamBundlesBundleDir = "bundles"
+)
 
-	repo := "clr-bundles/clr-bundles-" + ver + ".tar.gz"
-	if _, err := os.Stat(repo); err == nil {
-		fmt.Println("Already downloaded " + repo)
-		return
-	}
+func getUpstreamBundlesDir(ver string) string {
+	return filepath.Join(upstreamBundlesBaseDir, fmt.Sprintf(upstreamBundlesVerDir, ver), upstreamBundlesBundleDir)
+}
 
-	URL := "https://github.com/clearlinux/clr-bundles/archive/" + ver + ".tar.gz"
-	err := helpers.Download(repo, URL)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "ERROR: Failed to download clr-bundles, make sure the version is valid: %s\n", err)
-		os.Exit(1)
+func (b *Builder) getUpstreamBundles(ver string, prune bool) error {
```

Comment:
Optional: I'd consider making prune a separated function.

Optional (fallback): Unless you have some concrete local change that uses the "false" codepath, remove the argument and always assume true.

Reply:
>Optional: I'd consider making prune a separated function.

I considered that, but I didn't want the vast majority of functions to have to make two calls. (I wish Go supported default values for variables like Python does. Or even nil values for boolean primitives would allow default programming easier. I could always make it a variadic function to make it so folks don't have to pass the prune flag if they don't want). Bottom line, I'll pull `prune` out into a separate function when we hit a use case where people need to prune outside of fetching.

>Optional (fallback): Unless you have some concrete local change that uses the "false" codepath, remove the argument and always assume true.

The concrete example using false is the forthcoming "update my bundles to a new upstream version #", where you need to have the old CLR bundles and the new CLR bundles at the same time to create diffs.

Reply:
"concrete example using false is the forthcoming" -> Fair enough.

Another alternative is to create constants and take a flags numeric argument, so that call sites look like

    getUpstreamBundles(ver, Prune)

Since this code is in flux at the moment, I don't mind keeping as is.